docs/management/gcache-write-set-cache-encryption.md

```diff
@@ -0,0 +1,100 @@
+| `enc_mk_uuid` | The first Master Key or if Galera detects that the preamble is inconsistent, which causes a full GCache reset and a new Master Key is required, generates this UUID. This change avoids reuse of existing keys, which may be compromised, from the keyring.  |
+| `enc_fk_id` | The File Key ID used for GCache ring buffer encryption. Encrypted with the Master Key and stored in base64 encoding. |
+| `enc_crc` | The cyclic redundancy check (CRC) calculated from all encryption-related keys. |
+
+### Controlling encryption
+
+Encryption is controlled using the wsrep_provider_options. 
+
+| Variable name | Default value | Allowed values |
+|---|---|---|
+| [`gcache.encryption`](#gcacheencryption) | off | on/off |
+| [`gcache.encryption_cache_page_size`](#gcacheencryption_cache_page_size) | 32KB | 2-512 |
+| [`gcache.encryption_cache_size`](#gcacheencryption_cache_size) | 16MB | 2 - 512  |
+| [`allocator.disk_pages_encryption`](#allocatordisk_pages_encryption) | off | on/off |
+| [`allocator.encryption_cache_page_size`](#allocatorencryption_cache_page_size) | 32KB |  |
+| [`allocator.encryption_cache_size`](#allocatorencryption_cache_size) | 16MB |  |
+
+## Variable descriptions
+
+The following sections describe the variables related to Record-Set cache encryption.
+All variables are read-only.
+
+### gcache.encryption
+
+Enable or disable GCache encryption.
+
+### gcache.encryption_cache_page_size
+
+The size of the encryption cache for Record-Set pages. The value must be multiple of the CPU page size (typically 4kB). If the value is not, the server reports an error and stops.
+
+The maximum limit is 512. The minimum limit is 2.
+
+### gcache.encryption_cache_size
+
+The size of the encryption cache for GCache on-disk files. If `gcache.encryption=yes`, one encryption cache is associated with the GCache Ringbuffer file. Create an on-disk page and an encryption cache for that page is also created.
+
+The value cannot be smaller than the CPU page size, typically 4KB. If the value is smaller, the server reports an error and stops. We recommend a large cache since an extremely small encryption cache causes frequent encrypted pages swapping and degrade performance. 
+
+The `encryption_cache_size` should be a multiple of `encryption_cache_page_size`. If the value is not, the actual cache size is rounded up.
+
+The `encryption_cache_size` maximum is 512 allocation pages in the encryption cache. If the cache size must increase, you must also increase the `allocator.encryption_cache_page_size`.
+
+### allocator.disk_pages_encryption
+
+Enable or disable the allocator's disk pages encryption.
+
+### allocator.encryption_cache_page_size
```

Comment:
Here we start 2nd section: related to Write-Set cache encryption

Parameters are similar to ones in GCache section above, description is similar, the only difference is that now we are talking about Write-Set cahce not GCache

Reply:
Added.